projects/observability/src/shared/components/bar-gauge/bar-gauge.component.ts

```diff
@@ -0,0 +1,102 @@
+  selector: 'ht-bar-gauge',
+  styleUrls: ['./bar-gauge.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="bar-gauge">
+      <div *ngIf="this.title" class="title">{{ this.title | htDisplayTitle }}</div>
+      <div class="count">
+        {{ this.totalValue | number }} / {{ this.maxValue | number }}
+        <span class="units" *ngIf="this.units">{{ this.units }}</span>
+      </div>
+      <div class="bar">
+        <div #maxValueBar class="max-value-bar" [ngClass]="{ 'over-max-value': this.overMaxValue }">
+          <div #segmentBars class="segment-bars">
+            <div
+              *ngFor="let segment of this.barSegments"
+              class="segment-bar"
+              [ngClass]="{ 'hide-divider': this.nearMaxValue }"
+              [style.background]="segment.color"
+              [style.width.%]="segment.percentage"
+            >
+              <div class="divider"></div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `
+})
+export class BarGaugeComponent implements OnChanges {
+  @ViewChild('maxValueBar', { read: ElementRef })
+  public maxValueBar!: ElementRef;
+
+  @ViewChild('segmentBars', { read: ElementRef })
+  public segmentBars!: ElementRef;
+
+  @Input()
+  public title?: string;
+
+  @Input()
+  public units?: string;
+
+  @Input()
+  public maxValue?: number;
+
+  @Input()
+  public segments?: Segment[] = [];
+
+  public barSegments: BarSegment[] = [];
+  public totalValue: number = 0;
+  public overMaxValue: boolean = false;
+  public nearMaxValue: boolean = false;
+
+  public ngOnChanges(changes: TypedSimpleChanges<this>): void {
+    if (changes.segments || changes.maxValue) {
+      this.totalValue = this.calcTotalValueFromSegments(this.segments ?? []);
+      this.barSegments = this.toBarSegments(this.segments ?? [], this.maxValue ?? 0);
+
+      this.overMaxValue = this.maxValue !== undefined && this.totalValue > this.maxValue;
+
+      this.checkNearMaxValue();
+    }
+  }
+
+  private checkNearMaxValue(): void {
+    setTimeout(() => {
+      /*
+       * We want to remove the white divider if we fill up the bar. Compare segments to
+       * max bar width minus 2px for the divider width.
+       */
+      this.nearMaxValue = this.segmentBars.nativeElement.offsetWidth >= this.maxValueBar.nativeElement.offsetWidth - 2;
+    });
+  }
+
+  private toBarSegments(segments: Segment[], maxValue: number): BarSegment[] {
+    return segments.map(segment => this.toBarSegment(segment, getPercentage(segment.value, maxValue ?? 0)));
+  }
+
+  private toBarSegment(segment: Segment, percentage: number): BarSegment {
+    return {
+      ...segment,
+      percentage: percentage
+    };
+  }
+
+  private calcTotalValueFromSegments(segments: Segment[]): number {
+    return segments.reduce((previousValue, currentValue) => previousValue + currentValue.value, 0);
+  }
+}
+
+export interface Segment {
+  value: number;
+  color?: string;
```

Comment:
If color isn't provided, what are we defaulting to?

Reply:
$blue-7

<img width="194" alt="Screen Shot 2020-12-18 at 1 20 50 PM" src="https://user-images.githubusercontent.com/45181984/102662515-dbb2d580-4133-11eb-8b97-05a5aa765a93.png">

Reply:
Should we use color service instead to generate a palette so the segments are different colors?

Reply:
I think that makes sense once we have a use case for it.

Reply:
Eh, screw it. I'll just do this now. 👍